pinot-controller/src/main/java/com/linkedin/pinot/controller/helix/core/realtime/PinotLLCRealtimeSegmentManager.java

```diff
@@ -77,6 +79,7 @@
   private static final int KAFKA_PARTITION_OFFSET_FETCH_TIMEOUT_MILLIS = 10000;
   protected static final int STARTING_SEQUENCE_NUMBER = 0; // Initial sequence number for new table segments
   protected static final long END_OFFSET_FOR_CONSUMING_SEGMENTS = Long.MAX_VALUE;
+  private static final int NUM_LOCKS = 4;
```

Comment:
Use StripedLocks from Guava (https://github.com/google/guava/wiki/StripedExplained) if we extend number of locks. It supports exactly this use case and handles hashing names to locks